### PR TITLE
Added "editable" info to rules in REST API

### PR DESCRIPTION
**Pull request by kaikreuzer**
This was part of https://github.com/openhab/openhab-core/pull/1451 and has been separated as an independent PR that can already been reviewed and merged independently.

Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

File: bundles/org.openhab.core.automation.rest/src/main/java/org/openhab/core/automation/rest/internal/RuleResource.java [hunk 1]
```diff
@@ -121,6 +126,15 @@ public RuleResource( //
         this.ruleRegistry = ruleRegistry;
     }
 
+    @Reference(cardinality = ReferenceCardinality.OPTIONAL, policy = ReferencePolicy.DYNAMIC)
```
Comment (cweitkamp) [inline bundles/org.openhab.core.automation.rest/src/main/java/org/openhab/core/automation/rest/internal/RuleResource.java:129]: Is this really an optional reference? Isn't it mandatory and we can pass it via constructor injection?

File: bundles/org.openhab.core.automation.rest/src/main/java/org/openhab/core/automation/rest/internal/dto/EnrichedRuleDTOMapper.java [hunk 2]
```diff
@@ -20,13 +21,16 @@
...
  */
 public class EnrichedRuleDTOMapper extends RuleDTOMapper {
 
-    public static EnrichedRuleDTO map(final Rule rule, final RuleManager ruleEngine) {
+    public static EnrichedRuleDTO map(final Rule rule, final RuleManager ruleEngine,
+            final ManagedRuleProvider managedRuleProvider) {
         final EnrichedRuleDTO enrichedRuleDto = new EnrichedRuleDTO();
         fillProperties(rule, enrichedRuleDto);
         enrichedRuleDto.status = ruleEngine.getStatusInfo(rule.getUID());
+        enrichedRuleDto.editable = managedRuleProvider.get(enrichedRuleDto.uid) != null;
```
Comment (cweitkamp) [inline bundles/org.openhab.core.automation.rest/src/main/java/org/openhab/core/automation/rest/internal/dto/EnrichedRuleDTOMapper.java:33]: If the reference stays optional I prefer to see a `null`-check on the `managedRuleProvider` here.

```suggestion
        enrichedRuleDto.editable = managedRuleProvider.get(rule.getUID()) != null;
```

File: bundles/org.openhab.core.automation.rest/src/main/java/org/openhab/core/automation/rest/internal/RuleResource.java (hunk 1 above)
Comment (kaikreuzer) [reply bundles/org.openhab.core.automation.rest/src/main/java/org/openhab/core/automation/rest/internal/RuleResource.java]: Agree, can be mandatory.

**Comment by kaikreuzer** [comment]
Thanks @cweitkamp, I have addressed your comments.

**Pull request merged**